src/web/components/SharingPermission/BulkAddPermissions.tsx

```diff
@@ -42,70 +44,129 @@ export function BulkAddPermissions({
+  const watchDataProviderChecked = watch('dataProviderChecked');
 
   const handleSave = (data: BulkAddPermissionsForm) => {
-    // TODO: call onBulkAddSharingPermission with the appropriate data
+    onBulkAddSharingPermission(getCheckedParticipantTypeIds(data, participantTypes));
   };
 
-  const onToggleShowRecommendedParticipants = () => {
+  const onToggleViewRecommendedParticipants = () => {
     setShowRecommendedParticipants((prevToggle) => !prevToggle);
-    // TODO: show the actual participants
+    // TODO: show/hide the actual participants
   };
 
+  const bulkAddPermissionsMessage = (
+    <>
+      <b>{getRecommendationMessageFromTypeNames(currentParticipantTypeNames)}</b>
+      <p>
+        Allow all current and future participants within a participation type to decrypt your UID2
+        tokens.
+      </p>
+    </>
+  );
+
+  const viewRecommendedParticipantsButton = (
+    <button
+      type='button'
+      className='text-button bulk-add-permissions-view-recommendations-button'
+      onClick={onToggleViewRecommendedParticipants}
+    >
+      {showRecommendedParticipants ? 'Hide Participants' : 'View Participants'}
+    </button>
+  );
+
+  const participantTypeCheckboxes = (
+    <div className='participant-type-checkbox-section'>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('publisherChecked')}
+      />
+      <span className='checkbox-label'>Publisher</span>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('advertiserChecked')}
+      />
+      <span className='checkbox-label'>Advertiser</span>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('DSPChecked')}
+      />
+      <span className='checkbox-label'>DSP</span>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('dataProviderChecked')}
+      />
+      <span className='checkbox-label'>Data Provider</span>
+    </div>
+  );
+
+  const saveRecommendationsButton = (
+    <div className='bulk-add-permissions-footer'>
+      <button type='submit' className='primary-button bulk-add-permissions-submit-button'>
+        Save Recommendations
+      </button>
+    </div>
+  );
+
+  const recommendationContent = (
+    <div className='bulk-add-permissions'>
+      <div className='bulk-add-permissions-body'>
+        {bulkAddPermissionsMessage}
+        {participantTypeCheckboxes}
+        {viewRecommendedParticipantsButton}
+      </div>
+      {saveRecommendationsButton}
+    </div>
+  );
+
+  const collapsibleContent = (
+    <div className='bulk-add-permissions'>
+      <div className='bulk-add-permissions-body'>
+        {bulkAddPermissionsMessage}
+        {participantTypeCheckboxes}
+        {/* Change this condition once we have APIs to get the which types have been shared with */}
```

Comment:
hey @jingyi-gao-ttd, in an attempt to merge these changes in while blocked on the APIs, I have hardcoded the warning banner to never show, since the logic I had before was incorrect. The correct logic requires the APIs. Is that okay with you?

Reply:
That sounds reasonable to me